pkg/executors/shell_executor.go

```diff
@@ -17,23 +18,29 @@ import (
 
 type ShellExecutor struct {
 	Executor
-
-	Logger     *eventlogger.Logger
-	Shell      *shell.Shell
-	jobRequest *api.JobRequest
-
-	tmpDirectory string
+	Logger            *eventlogger.Logger
+	Shell             *shell.Shell
+	SelfHosted        bool
+	jobRequest        *api.JobRequest
+	tmpDirectory      string
+	cleanupAfterClose []string
 }
 
-func NewShellExecutor(request *api.JobRequest, logger *eventlogger.Logger) *ShellExecutor {
+func NewShellExecutor(request *api.JobRequest, logger *eventlogger.Logger, selfHosted bool) *ShellExecutor {
 	return &ShellExecutor{
-		Logger:       logger,
-		jobRequest:   request,
-		tmpDirectory: "/tmp",
+		Logger:            logger,
+		jobRequest:        request,
+		tmpDirectory:      os.TempDir(),
+		SelfHosted:        selfHosted,
```

Comment:
I would prefer here to have a feature that is enabled disabled, instead of setting `self_hosted` based on which we decide down in the code what is enabled.

Why this?

Compare these two constructors, where we enable something based on the price of the car for example. 

Implicit logic: 

``` golang
type Car struct {
  Expensive bool
}

func NewCar(expensive bool) *Price {
  return &Car{Expensive: expensive}
}

// ...
// then somewhere down in the code we do:

   if c.Expensive {
      c.startVoiceAssistant()    
   }

/// ... 
```

Explicit logic:

``` golang
type Car struct {
  hasVoiceAssistence bool
  hasHeating bool
  hasParkingRadar bool
  breaksAfter100kMiles bool
}

func NewCar(expensive bool) *Price {
  return &Car{
    hasVoiceAssistence: expensive,
    hasHeating: expensive,
    hasParkingRadar: expensive,
    breaksAfter100kMiles: !expensive,
  }
}

// ...
// then somewhere down in the code we do:

   if c.hasVoiceAssistence {
      c.startVoiceAssistant()    
   }

/// ... 
```

With this, the logic in the code below is very explicit and direct `if hasVoice => enableVoice` which makes it easier to understand.

Also, you have one place in the constructor where all the decisions that are related to the price are defined.